invoice/src/main/java/org/killbill/billing/invoice/InvoiceDispatcher.java

```diff
@@ -368,9 +369,13 @@ public Invoice apply(final InvoiceModelDao input) {
 
                 setChargedThroughDates(billingEvents.getAccountDateAndTimeZoneContext(), invoice.getInvoiceItems(FixedPriceInvoiceItem.class), invoice.getInvoiceItems(RecurringInvoiceItem.class), context);
 
-                // TODO we should send bus events when we commit the ionvoice on disk in commitInvoice
-                postEvents(account, invoice, adjustedUniqueOtherInvoiceId, isRealInvoiceWithNonEmptyItems, context);
+                if (InvoiceStatus.COMMITTED.equals(invoice.getStatus())) {
+                    // TODO we should send bus events when we commit the ionvoice on disk in commitInvoice
+                    postEvents(account, invoice, adjustedUniqueOtherInvoiceId, isRealInvoiceWithNonEmptyItems, context);
 
+                }
+
+                // TODO should we include this notification inside of previous if clause?
                 notifyAccountIfEnabled(account, invoice, isRealInvoiceWithNonEmptyItems, context);
```

Comment:
@matias-aguero-hs After talking to @sbrossie, there is indeed a cleanup task but let's not worry about it right now. However, the `notifyAccountIfEnabled` method should only be called for `COMMITTED` invoices (in the block above).